source/Octopus.Client/OctopusAsyncRepository.cs

```diff
@@ -26,63 +30,75 @@ public static IOctopusAsyncRepository CreateRepository(this IOctopusAsyncClient
+            BuiltInPackageRepository = new BuiltInPackageRepositoryRepository(this);
+            CertificateConfiguration = new CertificateConfigurationRepository(this);
+            Certificates = new CertificateRepository(this);
+            Channels = new ChannelRepository(this);
+            CommunityActionTemplates = new CommunityActionTemplateRepository(this);
+            Configuration = new ConfigurationRepository(this);
+            DashboardConfigurations = new DashboardConfigurationRepository(this);
+            Dashboards = new DashboardRepository(this);
+            Defects = new DefectsRepository(this);
+            DeploymentProcesses = new DeploymentProcessRepository(this);
+            Deployments = new DeploymentRepository(this);
+            Environments = new EnvironmentRepository(this);
+            Events = new EventRepository(this);
+            FeaturesConfiguration = new FeaturesConfigurationRepository(this);
+            Feeds = new FeedRepository(this);
+            Interruptions = new InterruptionRepository(this);
+            LibraryVariableSets = new LibraryVariableSetRepository(this);
+            Lifecycles = new LifecyclesRepository(this);
+            MachinePolicies = new MachinePolicyRepository(this);
+            MachineRoles = new MachineRoleRepository(this);
+            Machines = new MachineRepository(this);
+            Migrations = new MigrationRepository(this);
+            OctopusServerNodes = new OctopusServerNodeRepository(this);
+            PerformanceConfiguration = new PerformanceConfigurationRepository(this);
+            ProjectGroups = new ProjectGroupRepository(this);
+            Projects = new ProjectRepository(this);
+            ProjectTriggers = new ProjectTriggerRepository(this);
+            Proxies = new ProxyRepository(this);
+            Releases = new ReleaseRepository(this);
+            RetentionPolicies = new RetentionPolicyRepository(this);
+            Schedulers = new SchedulerRepository(this);
+            ServerStatus = new ServerStatusRepository(this);
+            Spaces = new SpaceRepository(this);
+            Subscriptions = new SubscriptionRepository(this);
+            TagSets = new TagSetRepository(this);
+            Tasks = new TaskRepository(this);
+            Teams = new TeamsRepository(this);
+            Tenants = new TenantRepository(this);
+            TenantVariables = new TenantVariablesRepository(this);
+            UserRoles = new UserRolesRepository(this);
+            Users = new UserRepository(this);
+            VariableSets = new VariableSetRepository(this);
+            Workers = new WorkerRepository(this);
+            WorkerPools = new WorkerPoolRepository(this);
+            ScopedUserRoles = new ScopedUserRoleRepository(this);
+            UserPermissions = new UserPermissionsRepository(this);
+        }
+
+        public static async Task<IOctopusAsyncRepository> Create(IOctopusAsyncClient client, SpaceContext spaceContext = null)
```

Comment:
A static method is used to initialize `OctopusAsyncRepository` as we can do async here e.g. to load the SpaceRootDocument